app/app/src/main/java/org/omecproject/up4/cli/FlowsStatsCommand.java

```diff
@@ -0,0 +1,108 @@
+/*
+ SPDX-License-Identifier: Apache-2.0
+ SPDX-FileCopyrightText: 2022-present Intel Corporation
+ */
+package org.omecproject.up4.cli;
+
+import com.google.common.collect.Maps;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.omecproject.up4.impl.Up4AdminService;
+import org.onlab.packet.Ip4Address;
+import org.onosproject.cli.AbstractShellCommand;
+import org.onosproject.net.behaviour.upf.UpfCounter;
+
+import java.util.Map;
+
+/**
+ * UP4 UE flow stats.
+ */
+@Service
+@Command(scope = "up4", name = "flow-stats",
+        description = "Read all UE flows stats")
+public class FlowsStatsCommand extends AbstractShellCommand {
+
+    private static final String SEPARATOR = "-".repeat(40);
+    private static final int SLEEP_TIME_MS = 200;
+
+    @Override
+    protected void doExecute() throws Exception {
+        Up4AdminService adminService = get(Up4AdminService.class);
+
+        Map<Ip4Address, UpfCounter> beforeDownlink = Maps.newHashMap();
+        adminService.getDownlinkFlows().forEach(dlUpfFlow -> {
+            beforeDownlink.put(dlUpfFlow.getSession().ueAddress(), dlUpfFlow.getCounter());
+        });
+
+        Map<Ip4Address, UpfCounter> beforeUplink = Maps.newHashMap();
+        adminService.getUplinkFlows().forEach(ulUpfFlow -> {
+            beforeUplink.put(ulUpfFlow.getTermination().ueSessionId(), ulUpfFlow.getCounter());
```

Comment:
Same here, how do we deal with multiple terminations for the same session?

Reply:
See comment above, we can aggregate per UE. Does it make sense to keep stats separate per application?

Reply:
Same answer as above